agent/util-scripts/pbench-tool-meister

```diff
@@ -0,0 +1,836 @@
+        self.tool_dir = tool_dir
+        self.screen_name = f"pbench-tool-{group}-{name}"
+        self.start_process = None
+        self.stop_process = None
+        self.post_process = None
+
+    def start(self):
+        """Creates the background `screen` process running the tool's "start"
+        operation.
+        """
+        if (
+            self.start_process is not None
+            or self.stop_process is not None
+            or self.post_process is not None
+        ):
+            self.logger.error(
+                "Tool(%s) has unexpected process already running", self.name
+            )
+            return
+        # screen -dm -L -S \"${screen_name}\" ${pbench_bin}/tool-scripts/${name} --${action} --dir=${tool_output_dir} ${tool_opts[@]}
+        args = [
+            "/usr/bin/screen",
+            "-dmS",
+            self.screen_name,
+            f"{self.pbench_bin}/tool-scripts/{self.name}",
+            "--start",
+            f"--dir={self.tool_dir}",
+            self.tool_opts,
+        ]
+        self.logger.info("%s: start_tool -- %s", self.name, " ".join(args))
+        self.start_process = subprocess.Popen(args)
+
+    def stop(self):
+        """Creates the background `screen` process to running the tool's "stop"
+        operation.
+        """
+        if self.start_process is None:
+            self.logger.error("Tool(%s)'s start process not already running", self.name)
+            return
+        if self.stop_process is not None:
+            self.logger.error(
+                "Tool(%s) has unexpected stop process already running", self.name
+            )
+            return
+        if self.post_process is not None:
+            self.logger.error(
+                "Tool(%s) has unexpected post-process process already" " running",
+                self.name,
+            )
+            return
+
+        # FIXME - before we "stop" a tool, check to see if a
+        # "{tool}/{tool}.pid" file exists.  If it doesn't wait for a second to
+        # show up, if after a second it does not show up, then give up waiting
+        # and just call the stop method.
+        tool_pid_file = self.tool_dir / self.name / f"{self.name}.pid"
+        cnt = 0
+        while not tool_pid_file.exists() and cnt < 100:
+            time.sleep(0.1)
+            cnt += 1
+        if not tool_pid_file.exists():
+            self.logger.warning(
+                "Toll(%s) pid file, %s, does not exist after waiting 10 seconds",
+                self.name,
+                tool_pid_file,
+            )
+
+        # $pbench_bin/tool-scripts/$name --$action --dir=${tool_output_dir} "${tool_opts[@]}"
+        args = [
+            f"{self.pbench_bin}/tool-scripts/{self.name}",
+            "--stop",
+            f"--dir={self.tool_dir}",
+            self.tool_opts,
+        ]
+        self.logger.info("%s: stop_tool -- %s", self.name, " ".join(args))
+        self.stop_process = subprocess.Popen(args)
+
+    def wait(self):
```

Comment:
Can we add a timeout here? Wait for x number of seconds/minutes timeout with an exception?

Reply:
The `stop` process has the logic to wait and then kill the process.

A follow-on commit will merge the `tool-scripts` into the Tool Meister class methods so that we won't have to spawn subprocesses for stopping like we do today.